app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -281,6 +285,54 @@ class UndertakingController @Inject() (
     }
   }
 
+  def getDisableUndertakingWarning: Action[AnyContent] = withCDSAuthenticatedUser.async { implicit request =>
+    withLeadUndertaking(undertaking => Ok(disableUndertakingWarningPage(undertaking.name.toString)).toFuture)
+  }
+
+  def getDisableUndertakingConfirm: Action[AnyContent] = withCDSAuthenticatedUser.async { implicit request =>
+    withLeadUndertaking(undertaking =>
+      Ok(disableUndertakingConfirmPage(disableUndertakingConfirmForm, undertaking.name)).toFuture
+    )
+  }
+
+  def postDisableUndertakingConfirm: Action[AnyContent] = withCDSAuthenticatedUser.async { implicit request =>
+    withLeadUndertaking { undertaking =>
+      disableUndertakingConfirmForm
+        .bindFromRequest()
+        .fold(
+          errors => BadRequest(disableUndertakingConfirmPage(errors, undertaking.name)).toFuture,
+          form => Redirect(getNext(form)).toFuture
+        )
+    }
+  }
+
+  def getUndertakingDisabled: Action[AnyContent] = withCDSAuthenticatedUser.async { implicit request =>
```

Comment:
Should we be modifying state on a `GET` request. I'd expect this to happen on the POST where the user submits the form that actions the disable with the following redirect just being a `GET` to a confirmation page.

Reply:
I think I initially Planned to do on POST, then changed it and now can't remember why I changed it 🤦‍♀️ . Let me try to remember otherwise  will change it back to POST

Reply:
No worries! Thanks for taking a look. :)

Reply:
Yes I remember now. If we disable  it on POST, then withLeadUndertaking will not return anything on  GET of getUndertakingDisabled and it will be redireted to first page " Do you claim custome duty waivers?". We need to show this page only to the lead that's why I checked it with withLeadUndertaking. If  this check doesn't matter then I can remove it. What do you think?

Reply:
Hmm...if we disable the undertaking does this then mean that an attempt to fetch the undertaking results in a 404? Either way if the undertaking is disabled does it matter whether the logged in user is the lead or not for this page? Might be worth clarifying the requirements if this is ambiguous. Makes sense to me to show this new page to any user when the undertaking is disabled.

Reply:
yeah thought so, If the page is not doing any processing then it's ok if some one try to access it using the URL

Reply:
I think so, yes.